Sources/Jobs/Scheduler/RecurrenceRule/ScheduleTimeUnit.swift

```diff
@@ -0,0 +1,92 @@
+import Foundation
+
+public enum ScheduleTimeUnit: CaseIterable {
+    case second
+    case minute
+    case hour
+    case day
+    case week
+    case month
+    case year
+}
+
+// inclusive
+func resolveValidUpperBound(_ ruleTimeUnit: RecurrenceRuleTimeUnit) -> Int? {
```

Comment:
These should be moved into some kind of object somewhere instead of being floating public methods